src/python/grpcio/grpc/_channel.py

```diff
@@ -212,7 +214,14 @@ def consume_request_iterator():
         )
         call.start_batch(cygrpc.Operations(operations), event_handler)
         state.due.add(cygrpc.OperationType.send_close_from_client)
-  thread = threading.Thread(target=consume_request_iterator)
+
+  def stop_write_thread(timeout):
+    write_thread_stopped = True
```

Comment:
I don't think this has the intended effect - I think this adds a new local field and assigns it to `True` rather than mutating the local field in the scope of `_consume_request_iterator`.

Also rather than adding new state I think what needs to happen here should just happen with the available state. Cancel the RPC if it hasn't already terminated, right?